sqlglot/executor/python.py

```diff
@@ -164,24 +181,38 @@ def join(self, step, context):
         condition = self.generate(step.condition)
         projections = self.generate_tuple(step.projections)
 
-        if not condition or not projections:
+        if not condition and not projections:
             return source_context
+        elif not projections:
```

Comment:
This all feels pretty messy to me.

Without projections, we need to keep the nested tables in the resulting context.

Reply:
i don't get this, you may have to explain it to me

Reply:
let's chat today